play-json/jvm/src/main/scala/play/api/libs/json/EnvReads.scala

```diff
@@ -528,6 +532,72 @@ trait EnvReads {
     }
   }
 
+  private def jdurationNumberReads(unit: TemporalUnit) =
+    Reads[JDuration] {
+      case JsNumber(n) if !n.ulp.isValidLong =>
+        JsError("error.invalid.longDuration")
+
+      case JsNumber(n) => JsSuccess(JDuration.of(n.toLong, unit))
+      case _ => JsError("error.expected.duration")
+    }
+
+  /**
+   * Deserializer of Java Duration from an integer (long) number,
+   * using the specified temporal unit.
+   */
+  def javaDurationNumberReads(unit: TemporalUnit): Reads[JDuration] =
+    jdurationNumberReads(unit)
+
+  /** Deserializer of Java Duration from a number of milliseconds. */
+  val javaDurationMillisReads: Reads[JDuration] =
+    javaDurationNumberReads(ChronoUnit.MILLIS)
+
+  /**
+   * Deserializer of Java Duration, from either a time-based amount of time
+   * (string representation such as '34.5 seconds'),
+   * or from a number of milliseconds (see [[javaDurationMillisReads]]).
+   */
+  implicit val DefaultJavaDurationReads: Reads[JDuration] = Reads[JDuration] {
+    case JsString(repr) => try {
+      JsSuccess(JDuration.parse(repr))
+    } catch {
+      case _: Exception => JsError("error.invalid.duration")
```

Comment:
I would consider it unsafe to catch an unexpected exception and map it to what looks like a validation error. This hides the true cause of the error from the caller.

If code like this throws an exception we don't already know about, it's likely because of either a bug or another serious issue on the server. I don't believe these things are our job to handle. `JsResult` is not a generic error handling mechanism; it's specifically for JSON validation. If you want to handle other exceptions, you should use a `Try`. With `JsResult.toTry`, you can do:

```scala
val result: Try[Foo] = Try(Json.fromJson[Foo](json)).flatMap(JsResult.toTry)
```

If we are trying to make the API safer, I would start with methods like `Json.parse`, and make sure we at least document the exceptions thrown if the JSON is invalid. Then we can start thinking about how to provide better ways for handling exceptions in general. But I think that is best done at a high level rather than being the responsibility of each individual `Reads`.